src/libs/actions/PersonalDetails/index.js

```diff
@@ -0,0 +1,290 @@
+            const details = participantEmails
+                .filter(login => !data[login])
+                .reduce((previousDetails, login) => ({
+                    ...previousDetails,
+                    [login]: {}, // Simply just need the key to exist
+                }), existingDetails);
+
+            const formattedPersonalDetails = formatPersonalDetails(details);
+            Onyx.merge(ONYXKEYS.PERSONAL_DETAILS, formattedPersonalDetails);
+
+            // The personalDetails of the participants contain their avatar images. Here we'll go over each
+            // report and based on the participants we'll link up their avatars to report icons.
+            const reportsToUpdate = {};
+            _.each(reports, (report) => {
+                if (report.participants.length > 0) {
+                    const avatars = _.map(report.participants, dmParticipant => ({
+                        firstName: lodashGet(details, [dmParticipant, 'firstName'], ''),
+                        avatar: lodashGet(details, [dmParticipant, 'avatar'], '') || getDefaultAvatar(dmParticipant),
+                    }))
+                        .sort((first, second) => first.firstName - second.firstName)
+                        .map(item => item.avatar);
+                    const reportName = _.chain(report.participants)
+                        .filter(participant => participant !== currentUserEmail)
+                        .map(participant => lodashGet(
+                            formattedPersonalDetails,
+                            [participant, 'displayName'],
+                            participant,
+                        ))
+                        .value()
+                        .join(', ');
+
+                    reportsToUpdate[`${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`] = {icons: avatars, reportName};
+                }
+            });
+
+            // We use mergeCollection such that it updates ONYXKEYS.COLLECTION.REPORT in one go.
+            // Any withOnyx subscribers to this key will also receive the complete updated props just once
+            // than updating props for each report and re-rendering had merge been used.
+            Onyx.mergeCollection(ONYXKEYS.COLLECTION.REPORT, reportsToUpdate);
+        });
+}
+
+
+/**
+ * Sets the personal details object for the current user
+ *
+ * @param {Object} details
+ */
+function setPersonalDetails(details) {
+    API.PersonalDetails_Update({details: JSON.stringify(details)});
+
+    if (details.timezone) {
+        NameValuePair.set(CONST.NVP.TIMEZONE, details.timezone);
+    }
+
+    // Update the associated onyx keys
+    Onyx.merge(ONYXKEYS.MY_PERSONAL_DETAILS, details);
+    Onyx.merge(ONYXKEYS.PERSONAL_DETAILS, formatPersonalDetails({[currentUserEmail]: details}));
+}
+
+/**
+ * Gets the preferred currency for the current user
+ *
+ * @param {Object} details
+ */
+function setCurrencyPreferences() {
```

Comment:
Perhaps this should be called `fetchCurrencyPreferences` to match the other functions called on app start.

Reply:
Sure!